src/btree/bt_read.c

```diff
@@ -151,15 +152,32 @@ __page_read(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
     if (prepare)
         WT_ERR(__wt_page_inmem_prepare(session, ref));
 
-skip_read:
     /*
      * In the case of a fast delete, move all of the page's records to a deleted state based on the
      * fast-delete information. Skip for special commands that don't care about an in-memory state.
+     *
+     * Note: there are three possible cases - the state was WT_REF_DELETED and ft_info.del was NULL;
+     * the state was WT_REF_DELETED and ft_info.del was non-NULL; and the state was WT_REF_DISK and
+     * the parent page cell was a WT_CELL_ADDR_DEL cell. The last is only valid in a readonly tree.
+     *
```

Comment:
Can we assert it is a readonly tree in this case?

Reply:
I thought it did... yeah, it already does. (?)